### PR TITLE
feat(chat client): Adding exploding lifetime option to send and attach.

### PR DESCRIPTION
**Pull request by jhoscar1**

### CODE REVIEW

File: src/chat-client/index.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ export interface ChatSendOptions {
   membersType?: chat1.ConversationMembersType
   confirmLumenSend?: boolean
   replyTo?: chat1.MessageID
+  explodingLifetime?: string
```
Comment (songgao) [inline src/chat-client/index.ts]: Thanks for the PR! I think `explodingLifetime` should be a number here (same for the other one), as in millisecond, just like how the duration is specified in other JS apis like `setTimeout`. Then before sending it to the CLI api, we just need to add the `ms` so they can be properly [parsed](https://golang.org/pkg/time/#ParseDuration) on the [Go side](https://github.com/keybase/client/blob/b9875b85992d7af0573cab8afb6aa46f7227997a/go/client/chat_api_handler.go#L167).

**Comment by jhoscar1** [comment]
@songgao thanks for the quick response! Will make these changes

**Comment by jhoscar1** [comment]
I wasn't certain if the options processing should happen in the formatAPIObjectInput function but it seemed like a good single point to place it @songgao

**Comment by pzduniak** [comment]
https://github.com/keybase/keybase-bot/blob/master/src/kvstore-client/index.ts#L75

I'd implement it like this instead, possibly do something like `explodingLifetime: options?.explodingLifetime ? `${options.explodingLifetime}ms` : undefined`. That API formatting code is ugly enough as-is.

**Comment by jhoscar1** [comment]
@pzduniak Good call. Thanks! Moved the suffix addition to the client methods.

File: src/utils/formatAPIObject.ts [hunk 2]
```diff
@@ -107,10 +103,7 @@ function matchBlacklist(context: FormatAPIObjectOutputContext | null): boolean {
  * @param key - The key to apprent to the parent array.
  * @returns - A new context.
  */
-function buildContext(
-  context: FormatAPIObjectOutputContext | null,
-  key: any
-): FormatAPIObjectOutputContext | null {
+function buildContext(context: FormatAPIObjectOutputContext | null, key: any): FormatAPIObjectOutputContext | null {
```
Comment (songgao) [inline src/utils/formatAPIObject.ts]: Mind revert whitespace changes in this file?

File: src/chat-client/index.ts [hunk 3]
```diff
@@ -309,7 +312,12 @@ class Chat extends ClientBase {
   ): Promise<chat1.SendRes> {
     await this._guardInitialized()
     const conv = this.getChannelOrConversationId(channelOrConversationId)
-    const args = {...options, ...conv, filename}
+    const args = {
+      ...options,
+      explodingLifetime: options.explodingLifetime ? `${options.explodingLifetime}ms` : null,
```
Comment (songgao) [inline src/chat-client/index.ts]: s/null/undefined. Same above.

File: src/utils/formatAPIObject.ts (hunk 2 above)
Comment (jhoscar1) [reply src/utils/formatAPIObject.ts]: @songgao I'm not sure what I'm doing wrong but eslint keeps overwriting these files

**Comment by songgao** [comment]
Thanks @jhoscar1 !

**Pull request merged**